mobile/android/android-virtual-device-service.ts

```diff
@@ -0,0 +1,283 @@
+		this.$childProcess.spawn(this.pathToEmulatorExecutable, ['-avd', imageIdentifier], { stdio: "ignore", detached: true }).unref();
+	}
+
+	private async getAvailableEmulatorsCore(): Promise<Mobile.IAvailableEmulatorsOutput> {
+		let result: ISpawnResult = null;
+		let devices: Mobile.IDeviceInfo[] = [];
+
+		if (this.pathToVirtualDeviceManagerExecutable && this.$fs.exists(this.pathToVirtualDeviceManagerExecutable)) {
+			result = await this.$childProcess.trySpawnFromCloseEvent(this.pathToVirtualDeviceManagerExecutable, ["list", "avds"]);
+			if (result && result.stdout) {
+				devices = this.parseListAvdsOutput(result.stdout);
+			}
+		} else {
+			devices = this.listAvdsFromDirectory();
+		}
+
+		return { devices, errors: result && result.stderr ? [result.stderr] : [] };
+	}
+
+	private async getRunningEmulatorData(runningEmulatorId: string, availableEmulators: Mobile.IDeviceInfo[]): Promise<Mobile.IDeviceInfo> {
+		const imageIdentifier = await this.getImageIdentifierFromRunningEmulator(runningEmulatorId);
+		const runningEmulator = this.$emulatorHelper.getEmulatorByImageIdentifier(imageIdentifier, availableEmulators);
+		if (!runningEmulator) {
+			return null;
+		}
+
+		this.$emulatorHelper.setRunningAndroidEmulatorProperties(runningEmulatorId, runningEmulator);
+
+		return runningEmulator;
+	}
+
+	private getImageIdentifierFromRunningEmulator(emulatorId: string): Promise<string> {
+		const match = emulatorId.match(/^emulator-(\d+)/);
+		const portNumber = match && match[1];
+		if (!portNumber) {
+			return Promise.resolve(null);
+		}
+
+		return new Promise<string>(resolve => {
+			let isResolved = false;
+			let output: string = "";
+			const client = net.connect(portNumber, () => {
+				client.write(`avd name${EOL}`);
+			});
+
+			client.on('data', data => {
+				output += data.toString();
+
+				const name = this.getImageIdentifierFromClientOutput(output);
+				// old output should look like:
+				// Android Console: type 'help' for a list of commands
+				// OK
+				// <Name of image>
+				// OK
+				// new output should look like:
+				// Android Console: type 'help' for a list of commands
+				// OK
+				// a\u001b[K\u001b[Dav\u001b[K\u001b[D\u001b[Davd\u001b...
+				// <Name of image>
+				// OK
+				if (name && !isResolved) {
+					isResolved = true;
+					resolve(name);
+				}
+
+				client.end();
+			});
+		});
+	}
+
+	@cache()
+	private get pathToVirtualDeviceManagerExecutable(): string {
+		if (this.androidHome) {
+			if (this.$fs.exists(this.pathToAvdManagerExecutable)) {
+				return this.pathToAvdManagerExecutable;
+			}
+
+			if (this.$fs.exists(this.pathToAndroidExecutable)) {
+				return this.pathToAndroidExecutable;
+			}
+		}
+
+		return null;
+	}
+
+	@cache()
+	private get pathToAvdManagerExecutable(): string {
+		if (this.androidHome) {
+			const avdManagerPath = path.join(this.androidHome, "tools", "bin", "avdmanager");
+			if (this.$hostInfo.isWindows) {
+				return `${avdManagerPath}.exe`;
```

Comment:
is this required? Node.js should be able to spawn .exe files without explicitly passing the extension (which is not valid for  .cmd extension for example).